apps/explorer/src/app/components/txs/details/tx-chain-event.tsx

```diff
@@ -20,45 +18,26 @@ interface TxDetailsChainEventProps {
  * Multiple events will relay the same data, from each validator, so that the
  * deposit/withdrawal can be verified independently.
  *
- * Design considerations so far:
- * - The ethereum address should be a link to an Ethereum explorer
- * - Sender and recipient are shown because they are easy
- * - Amount is not shown because there is no formatter by asset component
+ * There are so many chain events that the specific components have been broken
+ * out in to individual components. `getChainEventComponent` determines which
+ * is the most appropriate based on the transaction shape. See that function
+ * for more information.
  */
 export const TxDetailsChainEvent = ({
   txData,
   pubKey,
   blockData,
 }: TxDetailsChainEventProps) => {
+  const child = useMemo(() => getChainEventComponent(txData), [txData]);
```

Comment:
comment (non-blocking): this looks a bit weird at first, it sort of seems the same as:
```
const TxDetailsChainEvent = ({ txData, ...}) => {
  return. (
    <TableStuff>
       ...
       <ChainEventComponent txData={txData} />
       ...
     </TableStuff>
  )
}
```

just instead of react components you kind of use a function and manually check if the input for it has changed? i don't dislike it, just seemed a bit anti-reacty.

Reply:
Valid - I'll change that. This was a refactor from something messier, and I just didn't realise how close to just-a-react-component it got 😆

Reply:
Fixed - it's now a component